### ISSUE TITLE
Restarting machined API kills it forever

### ISSUE THREAD

**Issue by andrewrynhard** [talos-systems/talos#1095]
```
[I] ~/s/g/t/talos-demo master* (22c7fbe) $ osctl services machined-api restart
error starting service: rpc error: code = Unavailable desc = transport is closing
[I] ~/s/g/t/talos-demo master* (22c7fbe) $ osctl services
error listing services: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/system/init/init.sock: connect: no such file or directory"
```

**Comment by smira** [comment]
This is design issue, from Slack:

```
so getting to what we discussed a while ago, I was going to do following things:
1. Implement full set of APIs: start, stop, restart
2. Allow service to deny any of the APIs (or explicitly enable some) - to make sure users can't do random things to the services

So some services might implement their own restrictions on whether they can be stopped, started or restarted, we can default to 'deny all' and enable 'restart' for the services we feel might benefit for it
```

**Comment by andrewrynhard** [comment]
@smira I believe we fixed this didn't we? Can I close?

**Comment by smira** [comment]
yep, fixed in #1161
